flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/PartitionRequestQueue.java

```diff
@@ -97,6 +107,18 @@ public void run() {
 		});
 	}
 
+	void registerPeriodicFlush(NetworkSequenceViewReader reader, long flushTimeout) {
+		if (flushTimeout == 0) {
```

Comment:
This condition check can be done earlier in `ResultPartition` or `ResultSubpartition` stack?